lib/compilers/v8.ts

```diff
@@ -39,25 +55,173 @@ export class V8Compiler extends BaseCompiler {
     }
 
     override getIrOutputFilename(inputFilename: string, filters: ParseFiltersAndOutputOptions): string {
-        return this.filename(path.dirname(inputFilename) + '/code.asm');
+        return this.filename(path.join(path.dirname(inputFilename), v8AsmRedirectTargetFileName));
     }
 
     public override getOutputFilename(dirPath: string, outputFilebase: string, key?: any) {
-        let filename;
-        if (key && key.backendOptions && key.backendOptions.customOutputFilename) {
-            filename = key.backendOptions.customOutputFilename;
+        if (dirPath) {
+            return path.join(dirPath, v8AsmRedirectTargetFileName);
         } else {
-            filename = 'code.asm';
+            return v8AsmRedirectTargetFileName;
+        }
+    }
+
+    override optionsForFilter(filters: ParseFiltersAndOutputOptions, outputFilename: string) {
+        return [];
+    }
+
+    override async doBuildstep(command, args, execParams) {
+        delete execParams.env;
+        delete execParams.ldPath;
+        let result = await this.exec('cmake', ['-S', '..', '-B', '..'], execParams);
+        result = await this.exec('cmake', ['--build', '..'], execParams);
+        return this.processExecutionResult(result);
+    }
+
+    override getExecutableFilename(dirPath: string, outputFilebase: string, key?) {
+        let entryFile = 'example.mjs';
+        // Pick filename from the cmake source
+        const filenameVariableRegex = /set\(JAVASCRIPT_ENTRY_FILE(.*)\)/;
+        if (filenameVariableRegex.test(key.source)) {
+            const matches = key.source.match(filenameVariableRegex);
+            if (matches && matches[1]) {
+                entryFile = matches[1];
+                entryFile = entryFile.replace(/['"]/g, '').trim();
+            }
         }
 
         if (dirPath) {
-            return path.join(dirPath, filename);
+            return path.join(dirPath, entryFile);
         } else {
-            return filename;
+            return entryFile;
         }
     }
 
-    override optionsForFilter(filters: ParseFiltersAndOutputOptions, outputFilename: string) {
-        return [];
+    override async cmake(files, key) {
```

Comment:
There's an awful lot of code here that appears to be copied from the base implementation of `cmake`. I'd like to find a way we can reuse more of the base implementation by making some changes to the `cmake` call to run. Else the support burden of keeping all this copied code in sync with any changes to `cmake` will increase over time. Later I'll grab the code and do a file-to-file difference (I'm doing this review on the website currently).